lib/media/track/mediatrack.js

```diff
@@ -145,27 +151,35 @@ class MediaTrack extends Track {
   _end() {
     this._log.debug('Ended');
     if (this._dummyEl) {
-      this._detachElement(this._dummyEl);
+      this._detachElementInternal(this._dummyEl);
```

Comment:
Why do we need to create a separate `_detachElementInternal` method? So on `_end`, calling `_detachElementInternal` will not update internal properties such as `_attachments` and `_elShims` right?

Reply:
we need seperate function because `_detachElement` is no-op for `_dummyEl` as its not added to `this._attachments`
It was a bug that caused us to never detach  `_dummyEl`

Reply:
Ah good one.